lib/_http_client.js

```diff
@@ -32,9 +32,18 @@ function ClientRequest(options, cb) {
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
-    agent = defaultAgent;
+  } else if (agent == null) {
```

Comment:
@jasnell it looks to me that it is required to be non-strict to match `undefined` and `null`